File: Mathematics/Palindrome/palindrome.java
public class palindrome {
    
    static boolean isPal(int number){
        int rev = 0;

        int temp = number;
        while(temp != 0){
            int lastDigit = temp % 10;
            rev = rev*10 + lastDigit;
            temp = temp/10;
        }
        return rev== number;
    }

    public static void main(String[] args){
        int number = 1331;
        System.out.println(isPal(number));
    }
}

File: Mathematics/Palindrome/palindrome.cpp
#include<iostream>
using namespace std;

bool palindrome(int data){
    int result = 0; 
    int n = data;
    while(n > 0){
        int temp = n%10; 
        result = result*10 + temp; 
        n = n/10;
    }
    if(data == result)
        return true; 
    else 
        return false;
}

int main()
{
    cout<<palindrome(1221)<<endl;
    return 0;
}